pkg/pillar/containerd/containerd.go

```diff
@@ -500,8 +500,9 @@ func ctrExec(ctx context.Context, domainName string, args []string) (string, str
 		stdErr bytes.Buffer
 	)
 	cioOpts := []cio.Opt{cio.WithStreams(new(bytes.Buffer), &stdOut, &stdErr), cio.WithFIFODir(fifoDir)}
-	// exec-id for task.Exec can NOT be longer than 71 runes
-	process, err := task.Exec(ctx, fmt.Sprintf("%.20d-%.50s", rand.Int(), domainName), pspec, cio.NewCreator(cioOpts...))
+	// exec-id for task.Exec can NOT be longer than 71 runes, on top of that it has to match:
+	//   ^[A-Za-z0-9]+(?:[._-](?:[A-Za-z0-9]+))*$:
+	process, err := task.Exec(ctx, fmt.Sprintf("%.50s%.20d", domainName, rand.Int()), pspec, cio.NewCreator(cioOpts...))
```

Comment:
@rvs This is OK for now, but do we expect multiple tasks for the same app instance name?

The domainName is already extra unique; zedcloud ensures that the app instance name is unique and we add a ".<N>" where N is between 0-255 to make the domainName even more unique. (That is from the days before zedcloud was created hence I didn't know that the app instance names would be unique on the controller.)

We might have EVE tasks which have no app instance name hence no domainname, in which case we need to create a unique name.